### DISCUSSION TITLE
Changing location of gif

### DISCUSSION THREAD

**Discussion by Sarooshki** [bitbank2/AnimatedGIF#86] [Q&A] [answered]
I apologize if this is a basic question but I am relatively new to this and I am trying to test different things.
I am trying to display a gif on the LCD screen as well as different texts. my LCD screen is 240x240 and I created a gif that is 60x60 that I am trying to place on a specific location on my screen so I can put text around it.
How can I do that?
I am able to display gifs on my screen but I am not sure how to control location of the gif so that I can also display other things on there.

**Comment by bitbank2** [comment, marked as answer]
In the GIFDRAW structure is a void pointer (void *) called pUser. This is for you to pass whatever you want (pointer to a structure) to the GIFDraw callback function. It's the 3rd (optional) parameter for the playFrame() method.

playframe(bool bSync, int *delayMilliseconds, void *pUser);

Create a 2 element integer array with the corner offset you want and pass it as the pUser parameter.

**Reply by Sarooshki** [reply]
Thanks so much for getting back to me so quickly! I really appreciate it.
I am embarrassed to say that I do not quite understand the formatting of how this may look like. I tried your suggestion but I do not believe I did it correctly as the location did not change.
In your animated_gif_memory.ino you have the following code that outputs the gif.

```
  if (gif.open((uint8_t *)GIF_IMAGE, sizeof(GIF_IMAGE), GIFDraw))
  {
    Serial.printf("Successfully opened GIF; Canvas size = %d x %d\n", gif.getCanvasWidth(), gif.getCanvasHeight());
    tft.startWrite();
    while (gif.playFrame(true, NULL))
    {
      yield();
    }
    gif.close();
    tft.endWrite(); 
  }
```
In my code, I reformatted it so it looks as follows: 

```
  if (gif.open((uint8_t *)GIF_IMAGE, sizeof(GIF_IMAGE), GIFDraw))
  {
    Serial.printf("Successfully opened GIF; Canvas size = %d x %d\n", gif.getCanvasWidth(), gif.getCanvasHeight());
    tft.startWrite();
    //tft.setSwapBytes(true);
    //tft.pushImage(0,20,240,240,thermometer_resized);
    int gif_location[2] = {120,120};
    while (gif.playFrame(true, NULL,gif_location))
    {
      yield();
    }
  gif.close();
  tft.endWrite(); 
  }
```
I understand that you are referring to the function in the while loop gif.playFrame(true, NULL), and that I add the additional parameter to it. What I am confused on is how do I access the parameters of my gif. Earlier in the code, you declared "gif" as an AnimatedGIF class. We then included the header file of the gif and defined the gif as GIF_IMAGE. You used gif.open() and gif.playframe() from the AnimatedGIF class to do the output to the display, so what am I missing?

**Comment by bitbank2** [comment]
In GIFDraw:
int *pOffset = (int *)pDraw->pUser;
then use pOffset[0] and pOffset[1] as additional x/y offsets for drawing the pixels
